stuffListApi/public/controllers/controller.js

```diff
@@ -0,0 +1,176 @@
+var app = angular.module('myStuffApp', []);
+
+app.factory('Stuffs', ['$http', function($http) {
+    var stuffsFactory = {};
+
+    stuffsFactory.getStuffs = function() {
+        return $http.get('/stuffs');
+    }
+
+    stuffsFactory.createStuff = function(body) {
+        return $http.post('/stuffs', body);
+    }
+
+    stuffsFactory.readStuff = function(stuffId) {
+        return $http.get('/stuffs/' + stuffId);
+    }
+
+    stuffsFactory.updateStuff = function(stuffId, body) {
+        return $http.put('/stuffs/' + stuffId, body);
+    }
+
+    stuffsFactory.deleteStuff = function(stuffId) {
+        return $http.delete('/stuffs/' + stuffId);
+    }
+
+    stuffsFactory.deleteAllStuff = function() {
+        return $http.delete('/stuffs');
+    }
+
+    return stuffsFactory;
+}]);
+
+app.controller('StuffController', ['$scope', '$filter', 'Stuffs', function($scope, $filter, Stuffs) {
+    $scope.currentStuffId = null;
+
+    var showSpinner = function(displaySpinner) {
+        if (displaySpinner) {
+            $('#spinner').show();
+        } else {
+            $('#spinner').hide();
+        }
+    };
+
+    var refresh = function() {
+        Stuffs.getStuffs().then(function successCallback(response) {
+            $scope.stuffs = response.data;
+            putRecordInCategory();
+            showSpinner(false);
+        }, function errorCallback(response) {
+            console.log(response.data);
+            showSpinner(false);
+        });
+    };
+
+    var readRecord = function(stuffId, modalTitle, action) {
+        var record = null;
+        Stuffs.readStuff(stuffId).then(function successCallback(response) {
+            record = response.data;
+            $scope.stuffName = record.name;
+            $scope.stuffCategory = record.category;
+            $scope.currentStuffId = record._id;
+            genericModalConfig(modalTitle, action);
+        }, function errorCallback(response) {
+            console.log(response.data);
+        });
+    };
+
+    var putRecordInCategory = function() {
+        var categories = [];
+        $scope.stuffCategorized = categories;
+
+        for (var i = 0; i < $scope.stuffs.length; i++) {
+            if (categories.length) {
+                var match = false;
+                var counter = categories.length;
+                for (var j = 0; j < counter; j++) {
```

Comment:
Very nice article, actually It was a quick approach, I was planning to do this on the backend and get all the _stuffs_ with their corresponding categories, but for the timing, I made it on the UI. This past weekend I had a release to production so I was working from home and in parallel working on this app, thanks for your comment, it is very helpful :+1: .